### ISSUE TITLE
Include resampler latency in cubeb_stream_get_latency?

### ISSUE THREAD

**Issue by mdsitton** [kinetiknz/cubeb#527]
From what i can tell cubeb_resampler_latency() is not currently used anywhere in the code. For my usage of the library i need to be aware of all sources of latency, and in some cases the resampler is one of those sources.

Having the latency induced by the resampler automatically included into the value returned by cubeb_stream_get_latency() would make things much simpler since the resampler doesn't seemed to be used on all platform implementations.

**Comment by achronop** [comment]
Yeah, that makes sense and it is easy to add it in `cubeb_stream_get_latency()`. Windows and OSX that need it most. The Android backend makes use of the resampler but it is usually configured in a way that it is not used unless you are doing extreme things with it.

**Comment by mdsitton** [comment]
I could probably work on making a pull request myself for this if you guys don't mind?

**Comment by kinetiknz** [comment]
That'd be great, thank you!

**Comment by mdsitton** [comment]
After investigating this change somewhat I've come across a deeper issue with the cubeb_stream_get_latency for wasapi atleast. IAudioClient::GetStreamLatency() seems to always return 0 on windows 10?
I've not found any documentation on this anywhere, other than like a single person complaining about it on the msdn forums with no response a couple years back. I've also tested #530 thinking maybe something with IAudioClient3 would fix that but it doesn't.

Which after further digging i came across using the IAudioClock interface which is actually used already in the implementation of wasapi_stream_get_position(), with current_stream_delay()

So if no one object's here i'm planning on replacing the implementation of wasapi_stream_get_latency with something based on IAudioClock, along with adding any resampler frames. Should be able to get this up tomorrow.

**Comment by ligfx** [comment]
@mdsitton yep, I can reproduce that issue with GetStreamLatency on Windows 10 (always returns 0). If IAudioClient3 is being used (anywhere on the system) then GetCurrentSharedModeEnginePeriod will return the current latency. Otherwise it just returns the default latency (about 10ms) and not the size of the stream's buffer.
